test/queue_list_test.dart

```diff
@@ -258,6 +258,25 @@ void main() {
     expect(queue.cast<Pattern>(), same(queue));
   }, skip: isDart2 ? false : 'Requires a Dart2 runtime');
 
+  test("cast does not throw on mutation when the type is valid", () {
+    var patternQueue = new QueueList<Pattern>()..addAll(['a', 'b']);
+    QueueList<Object> untypedQueue = patternQueue;
+    var stringQueue = untypedQueue.cast<String>();
+    stringQueue.addAll(['c', 'd']);
```

Comment:
If you forgot to do the `cast`, the test would still succeed (either because it infers `<Pattern>['c', 'd']`, or because `List<String>` is-a `List<Pattern>`).

At least add `expect(stringQueue is List<String>, true);`, which shows that the `cast` actually did *something* (and skip in non-dart2 mode because there `cast` actually doesn't do anything).

Reply:
Done.